api/include/opentelemetry/metrics/noop.h

```diff
@@ -0,0 +1,137 @@
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace metrics {
+/**
+ * No-op implementation of Meter. This class should not be used directly.
+ */
+class NoopMeter : public Meter
+{
+public:
+  NoopMeter() = default;
+
+  // All NewInstrument functions return a shared_ptr to a no-op form of that instrument.
+
+  opentelemetry::nostd::shared_ptr<SynchronousInstrument> NewDoubleCounter(
+      nostd::string_view name,
+      nostd::string_view description,
+      nostd::string_view unit,
+      const bool enabled) override
+  {
+    return nostd::shared_ptr<SynchronousInstrument>{
+        new NoopDoubleCounter(name, description, unit, enabled)};
```

Comment:
Where is `NoopDoubleCounter` defined?

Reply:
This is defined in PR #161 written by @ankit-bhargava. I have tested my code with his privately to ensure compatibility. After his PR is merged I will rebase and, if necessary, update my code.